### DISCUSSION TITLE
Not able to list Objects in existing bucket

### DISCUSSION THREAD

**Discussion by samyswam** [juicedata/juicefs#3928] [General]
since we are using amzon linux the provided binary wont run so

step 1)i clone the branch and buit binary using go build commad.

step 2) juicefs -V
juicefs version 1.1.0-dev+unknown

step 3)Ec2 is already having IAM role
command to format
juicefs format --storage s3 --bucket https://halodoc-stage-heap-dump.s3.ap-southeast-1.amazonaws.com/ sqlite3://test.db tempstore
output
 <INFO>: Meta address: sqlite3://test.db [interface.go:497]
<INFO>: Data use s3://halodoc-stage-heap-dump/tempstore/ [format.go:471]
Failed to delete: AccessDenied: Access Denied
	status code: 403, request id: NZN79CQQ4213FJ0Q, host id: hjskEszjkZNhoocV3p9zuIupCFuG0oDiuSHsVwMiA/+7UphMp3XBNmMNxdBCzxMkxqfMgq5Ztjk=2023/07/14 08:13:41.184491 juicefs[19151] <INFO>: Volume is formatted as {
  "Name": "tempstore",
  "UUID": "64a68d92-091c-4ea5-b0fa-890528604fa7",
  "Storage": "s3",
  "Bucket": "https://halodoc-stage-heap-dump.s3.ap-southeast-1.amazonaws.com/",
  "BlockSize": 4096,
  "Compression": "none",
  "EncryptAlgo": "aes256gcm-rsa",
  "TrashDays": 1,
  "MetaVersion": 1,
  "MinClientVersion": "1.1.0-A",
  "DirStats": true
} [format.go:508]

step 4)  sudo juicefs mount -d sqlite3://test.db  /mnt/jfs
 <INFO>: Meta address: sqlite3://test.db [interface.go:497]
<INFO>: Data use s3://halodoc-stage-heap-dump/tempstore/ [mount.go:605]
2023/07/14 08:15:49.041203 juicefs[25648] <INFO>: Disk cache (/var/jfsCache/64a68d92-091c-4ea5-b0fa-890528604fa7/): capacity (102400 MB), free ratio (10%), max pending pages (15) [disk_cache.go:114]
2023/07/14 08:15:49.542462 juicefs[25648] <INFO>: OK, tempstore is ready at /mnt/jfs [mount_unix.go:48]

step 5)This bucket has already objects in it when i got /mnt/jfs abd do ls, it shows 0 results why????

df-h
JuiceFS:tempstore        1.0P  4.0K  1.0P   1% /mnt/jfs

Our bucket is private but has get,list,put access from our ec2

**Comment by SandyXSD** [comment]
Everything works as expected.
step 3) When formatting a new volume, JuiceFS will do a simple test (PUT/GET/DELETE an object) to make sure you can access the bucket. As DELETE is not a must here, you got the warning message but still successfully created the volume.
step 5) The volume is fresh new, so you got 0 result. The objects under the bucket are for internal usage (such as uuid) and metadata [backups](https://juicefs.com/docs/community/metadata_dump_load/#backup-automatically). All objects for data usage will have common prefix "chunks".

**Comment by samyswam** [comment]
But when i tried creating file in this path it is not reflecting in my s3

**Comment by samyswam** [comment]
If files ar stored in dynamic chunck names picked by juicefs its very difficult to identify a particular file is there any way to specify path/ file name option 
https://juicefs.com/docs/community/architecture/#how-juicefs-store-files

**Comment by SandyXSD** [comment]
Nope
